src/Plugin/Field/FieldWidget/OgComplex.php

```diff
@@ -26,13 +30,157 @@
 class OgComplex extends EntityReferenceAutocompleteWidget {
 
   /**
-   * The OG complex widget have a special logic on order to return the groups
-   * that user can reference to.
+   * {@inheritdoc}
    */
   public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
     $parent = parent::formElement($items, $delta, $element, $form, $form_state);
+    // todo: fix the definition in th UI level.
     $parent['target_id']['#selection_handler'] = 'default:og';
```

Comment:
not exactly here. This `formElement` is our container for the two child widgets (i.e. "my groups" and "other groups"). When creating any child widget, then we should set the correct selection handler settings, so we show the correct options per "field_mode"